### PR TITLE
Fix NxTreeViewMultiSelect focus states RSC-629

### PR DESCRIPTION
**Pull request by avnvbhatta**
https://issues.sonatype.org/browse/RSC-629

This PR fixes the incorrectly positioned focus states for `NxTreeViewMultiSelect`

- Previously, the `::after` pseudoelement was added to the `<label>`. If the label had some padding (which is what the `NxTreeViewMultiSelect` labels have), the alignment of the focus border is misaligned.
- This fix adds the `::after` pseudoelement to a `<span>` which doesn't change based on the label padding.
- I had to add a `<span>` on `NxRadio`, because initially the `::after` was being added to an `<svg>`. Since `<svg>` does not support generated content like `::after`, I had to create a `<span>` for the `::after `element to show

### CODE REVIEW

File: lib/src/components/NxCheckbox/NxCheckbox.scss [hunk 1]
```diff
@@ -16,6 +16,14 @@ $nx-checkbox-width: 16px;
   position: relative;
   width: $nx-checkbox-width;
 
+  // Pseudoelemtn style is for the focus border
```
Comment (glenhunter) [inline lib/src/components/NxCheckbox/NxCheckbox.scss:19]:
```suggestion
  // Pseudo-element style is for the focus border
```

File: lib/src/scss-shared/_nx-focus-border-helpers.scss [hunk 2]
```diff
@@ -9,9 +9,8 @@
 // Border size is 1px border and 6px here represents total space in each axis
 // between border and control elements. Values are based on RSC design spec.
 
-// NxRadio and NxCheckbox require the left property for correct positioning of focus border
+// NxRadio and NxCheckbox require the left and top property for correct positioning of focus border
```
Comment (glenhunter) [inline lib/src/scss-shared/_nx-focus-border-helpers.scss:12]: You say "left and top property" but I only see height and width below

File: lib/src/base-styles/_nx-radio-checkbox.scss [hunk 3]
```diff
@@ -32,12 +32,18 @@
...
-    &::after {
-      border: 1px solid var(--nx-color-interactive-border-focus);
-      box-shadow: var(--nx-box-shadow-focus);
-      content: '';
-      display: block;
-      position: absolute;
+    // For radios, .nx-radio-checkbox__control is added to an <svg>
+    // Since <svg> does not support pseudoelements, it is instead added to
+    // an additional <span> with the class nx-radio__focus
+    .nx-radio-checkbox__control, .nx-radio__focus {
```
Comment (glenhunter) [inline lib/src/base-styles/_nx-radio-checkbox.scss:38]: This combination of classes results in the attributes being applied to both the radio's SVG and `span`. I think that you need to replace `nx-radio-checkbox__control` with `nx-checkbox__box`

File: lib/src/components/NxCheckbox/NxCheckbox.scss [hunk 4]
```diff
@@ -16,6 +16,14 @@ $nx-checkbox-width: 16px;
...
+  &::after{
+    @include nx-radio-checkbox-focus-border($nx-checkbox-height, $nx-checkbox-width);
+    left: -5px;
+    top: -5px;
+    border-radius: 4px;
```
Comment (glenhunter) [inline lib/src/components/NxCheckbox/NxCheckbox.scss:24]:
```suggestion
    @include nx-radio-checkbox-focus-border($nx-checkbox-height, $nx-checkbox-width);
    
    border-radius: 4px;
    top: -5px;
    left: -5px;
```

Comment (glenhunter) [reply lib/src/components/NxCheckbox/NxCheckbox.scss:24]: Pedantic I know

File: lib/src/components/NxRadio/NxRadio.scss [hunk 5]
```diff
@@ -36,9 +36,12 @@ $nx-radio-width: 16px;
...
 }
 
-.nx-radio:focus-within {
+// Pseudoelement style is for the focus border
+.nx-radio__focus {
   &::after{
     @include nx-radio-checkbox-focus-border($nx-radio-height, $nx-radio-width);
+    left: -28px;
+    top: -12px;
     border-radius: 12px;
```
Comment (glenhunter) [inline lib/src/components/NxRadio/NxRadio.scss:45]:
```suggestion

    border-radius: 12px;
    top: -12px;
    left: -28px;
```

Comment (glenhunter) [reply lib/src/components/NxRadio/NxRadio.scss:45]: More pedantry

File: lib/src/components/NxCheckbox/NxCheckbox.scss (hunk 1 above)
Comment (avnvbhatta) [reply lib/src/components/NxCheckbox/NxCheckbox.scss]: > 

Fixed. 
a0345b6aad185d6724dce06df90bc4049d654dcd

File: lib/src/scss-shared/_nx-focus-border-helpers.scss (hunk 2 above)
Comment (avnvbhatta) [reply lib/src/scss-shared/_nx-focus-border-helpers.scss]: > You say "left and top property" but I only see height and width below

The `left` and `top` are currently specified in `NxRadio` and `NxCheckbox`'s respective .scss files
c64660e53c501a27168f5b17d48bf49041aa8cb3

File: lib/src/base-styles/_nx-radio-checkbox.scss (hunk 3 above)
Comment (avnvbhatta) [reply lib/src/base-styles/_nx-radio-checkbox.scss]: > This combination of classes results in the attributes being applied to both the radio's SVG and `span`. I think that you need to replace `nx-radio-checkbox__control` with `nx-checkbox__box`

This has been fixed.
ef22d2ecd2cc3eb0b8b7e1804cf7a32fcf4a76e4

File: lib/src/components/NxCheckbox/NxCheckbox.scss (hunk 4 above)
Comment (avnvbhatta) [reply lib/src/components/NxCheckbox/NxCheckbox.scss]: > Pedantic I know

Fixed.
a0345b6aad185d6724dce06df90bc4049d654dcd

File: lib/src/components/NxRadio/NxRadio.scss (hunk 5 above)
Comment (avnvbhatta) [reply lib/src/components/NxRadio/NxRadio.scss]: > More pedantry

More fixes! 😄 
a0345b6aad185d6724dce06df90bc4049d654dcd

**Review by glenhunter — APPROVED** [review]

**Pull request merged**